### ISSUE TITLE
Adding `rem` methods matching `divrem`

### ISSUE THREAD

**Issue by fingolfin** [Nemocas/AbstractAlgebra.jl#736]
We have `divrem` and `div`, so IMHO we should also have `rem`; then we'd be conforming with the Julia documentation for `divrem`.

The one "drawback" of this I can think of is that the `%` operator would work for our types, but that doesn't seem like a bad problem?

Of course the Julia documentation for `div`, `divrem`, `rem` is suboptimal for our purposes right now (too much focused on integers), but I am confident that we can get PRs to Julia merged which adjust them to better suite us.

**Comment by wbhart** [comment]
There are numerous AbstractAlgebra/Nemo tickets on this and a long backstory. It is unfortunately more or less impossible to make div, divrem and rem consistent as per Julia.

I spent an enormous amount of time trying to sort this all out when I started writing Oscar documentation. After a huge amount of discussions, many tickets, lots of breaking changes, we eventually managed to close out those tickets.

Here is a starting point for this issue, but there are many more:

https://github.com/Nemocas/Nemo.jl/issues/699

I quote from that ticket: "We must continue to use mod and our hacked versions of div and divrem everywhere. We are just going to have to explain this to users of Oscar and put up with endless tickets and revert wars in the distant future between developers who think it should be "fixed"."

There are some open tickets related to the behaviour of mod over Q (which seems similarly problematic) and other related issues. I don't see any way to resolve all these consistency issues, which is one reason I gave up.
